File: src/main/java/org/kite9/diagram/common/elements/vertex/SingleElementVertex.java
package org.kite9.diagram.common.elements.vertex;

import org.kite9.diagram.model.DiagramElement;

/**
 * Vertex which is part of a representing a single diagram element 
 * (e.g. part of a connection)
 * @author robmoffat
 *
 */
public interface SingleElementVertex extends Vertex {

	public DiagramElement getOriginalUnderlying();
	
}

File: a/src/jvmMain/kotlin/org/kite9/diagram/common/elements/vertex/SingleElementVertex.kt
package org.kite9.diagram.common.elements.vertex

import org.kite9.diagram.model.DiagramElement

/**
 * Vertex which is part of a representing a single diagram element
 * (e.g. part of a connection)
 * @author robmoffat
 */
interface SingleElementVertex : Vertex {

    fun getOriginalUnderlying(): DiagramElement
}